ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/helpers/LikeEvaluator.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.codegen.helpers;
+
+import com.google.common.collect.ImmutableSet;
+import io.confluent.ksql.util.KsqlException;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * Used in the code generation to evaluate SQL 'LIKE' expressions.
+ *
+ * @see io.confluent.ksql.execution.expression.tree.LikePredicate
+ */
+public final class LikeEvaluator {
+
+  // characters that need to be escaped within regex [] - all other
+  // characters are escaped using \Q \E (see Pattern#quote)
+  private static final Set<Character> ESCAPED_CHARS = ImmutableSet.of(
+      '[', ']', '^', '\\'
+  );
+
+  private LikeEvaluator() {
+  }
+
+  /**
+   * @param val       the value to match
+   * @param pattern   the pattern to match against
+   *
+   * @return  whether or not {@code val} matches {@code pattern} given the SQL
+   *          LIKE definition of matching
+   */
+  public static boolean matches(final String val, final String pattern) {
+    // note that we do not yet support escape characters in the pattern
+    // see issue
```

Comment:
missing issue link?

Reply:
https://github.com/confluentinc/ksql/issues/5021 will add it